src/libvirt-xml-parse.js

```diff
@@ -30,6 +30,60 @@ export function getDiskElemByTarget(domxml, targetOriginal) {
     }
 }
 
+export function getHostDevElemBySource(domxml, source) {
+    const domainElem = getElem(domxml);
+
+    if (!domainElem) {
+        console.warn(`Can't parse dumpxml, input: "${domainElem}"`);
+        return;
+    }
+
+    const devicesElem = domainElem.getElementsByTagName('devices')[0];
+    const hostdevElems = devicesElem.getElementsByTagName('hostdev');
+
+    if (hostdevElems) {
+        for (let i = 0; i < hostdevElems.length; i++) {
+            const hostdevElem = hostdevElems[i];
```

Comment:
That owuld break the code below, since when we find correct hostdevElem, we return it `return new XMLSerializer().serializeToString(hostdevElem);`
So I would have to create a tmp variable, put it outside of `forEach(() => {...});, `and return it at the end, at which point I think it's simpler to have the for... cycle

Reply:
Right, I was missing that, thanks.